### PR TITLE
don't crash on PyPy 7.0.0

### PR DESCRIPTION
**Pull request by antocuni**
PyPy 7.0.0 uses 'py3-style' rlocks, i.e. they are implemented natively instead of being in pure-python. Fix commit fixes the code in the sense that at least doesn't crash on PyPy, but it still suffers of issue546, as python3.6 does

### CODE REVIEW

**Comment by gordol** [comment]
any chance this can be merged?

**Comment by gordol** [comment]
we really need this for latest pypy support.

**Review by temoto — APPROVED** [review]
LGTM

File: eventlet/patcher.py [hunk 1]
```diff
@@ -353,17 +353,26 @@ def _green_existing_locks():
...
+        pyrlock_type = None
+
     # We're monkey-patching so there can't be any greenlets yet, ergo our thread
     # ID is the only valid owner possible.
     tid = eventlet.green.thread.get_ident()
     for obj in gc.get_objects():
         if isinstance(obj, rlock_type):
-            if (sys.version_info[0] == 2 and
+            if (not py3_style and
                     isinstance(obj._RLock__block, lock_type)):
```
Comment (temoto) [inline eventlet/patcher.py:373]:
```suggestion
            if not py3_style and isinstance(obj._RLock__block, lock_type):
```

File: eventlet/patcher.py [hunk 2]
```diff
@@ -353,17 +353,26 @@ def _green_existing_locks():
...
                 _fix_py2_rlock(obj, tid)
-            elif (sys.version_info[0] >= 3 and
+            elif (py3_style and
                     not isinstance(obj, pyrlock_type)):
```
Comment (temoto) [inline eventlet/patcher.py:376]:
```suggestion
            elif py3_style and not isinstance(obj, pyrlock_type):
```

**Review by jstasiak — APPROVED** [review]

**Pull request merged**